### ISSUE TITLE
Yield farming on ALEX

### ISSUE THREAD

**Issue by fiftyeightandeight** [alexgo-io/alex-v1#13]
A high LTV (even on a static collateral) can be offered between ayTokens of different maturities. For example, ayToken-1y as collateral may achieve 90% LTV to mint ayToken-O/N (similar to borrowing overnight to buy long-term asset).
This allows "yield farming" whereby you repeat the process of buying ayToken-1y, use it as collateral to mint and sell ayToken-O/N for Token (to buy ayToken-1y) to achieve a high APY (assuming ayToken-1y APY > ayToken-O/N APY).

For example,
1. Rachel has 100 BTC.
2. Rachel spends 100 BTC and buys ayBTC-1y. She uses it as collateral to borrow ayBTC-O/N at 90% LTV.
3. She sells the borrowed ayBTC-O/N (from (2)) and buys ayBTC-1y. She uses it as collateral to borrow ayBTC-O/N at 90% LTV.
4. She repeats (3).

If 1y APY is 10% and O/N APY is 5%, the above "yield farming" results in max APY of 55%.

Mathematically,

<img src="https://latex.codecogs.com/svg.latex?r=\frac{1+r_{1y}}{1-LTV}-\left(1+\frac{r_{1d}}{365}\right)^{365}\left(\frac{LTV}{1-LTV}\right)-1">

An illustration is available [here](https://docs.google.com/spreadsheets/d/1L-52KHFl7O_h22Fg4gpZKczdPEXuAt5yAh2gX3BQP58?authuser=alexd%40alexgo.io&usp=drive_fs).

**Comment by jingstat** [comment]
I wonder if there are any conditions on the limit of repeated collaterals, e.g. once per day?  Based on the illustration, if we can do an infinite number of loans within a day, the APY is exactly 55%  APY_1y/ (1-LTV) + APY_1d* LTV/(1-LTV).  Did I miss anything?

**Comment by Rredlion** [comment]
There are a lot of assumptions to be made to reach your conclusion. I think max 55% is right. but as you are exposed to almost 10x leverage, all the positions could be unwound, and you would lose the capital should margin falls below 90%

**Comment by fiftyeightandeight** [comment]
Yes, we will need to determine the max number of repeats a borrower can do and aggregate them as a single (atomic) transaction (so a single click allows this to happen). This will be an interesting alternative to liquidity miners (who look for high APY with principal at risk) and encourages liquidity build-up on the platform (10x exposure on dollar).
Thinking about offering this as a “pro” user option (given its risk).
________________________________
